nixos/modules/services/web-servers/meguca.nix

```diff
@@ -74,47 +80,56 @@ in
     security.sudo.enable = cfg.enable == true;
     services.postgresql.enable = cfg.enable == true;
 
+    services.meguca.passwordFile = mkDefault (toString (pkgs.writeTextFile {
+      name = "meguca-password-file";
+      text = cfg.password;
+    }));
+
+    services.meguca.postgresArgsFile = mkDefault (toString (pkgs.writeTextFile {
+      name = "meguca-postgres-args";
+      text = cfg.postgresArgs;
+    }));
+
     systemd.services.meguca = {
       description = "meguca";
       after = [ "network.target" "postgresql.service" ];
       wantedBy = [ "multi-user.target" ];
+      script = "/run/meguca/start";
 
       preStart = ''
         # Ensure folder exists and links are correct or create them
-        mkdir -p ${cfg.baseDir}
-        ln -sf ${pkgs.meguca}/share/meguca/www ${cfg.baseDir}
-        chown -R meguca:meguca ${cfg.baseDir}
+        mkdir -p /run/meguca
+        ln -sf ${pkgs.meguca}/share/meguca/www /run/meguca
 
         # Ensure the database is correct or create it
-        ${pkgs.sudo}/bin/sudo -u ${postgres.superUser} ${postgres.package}/bin/createuser -SDR meguca || true
-        ${pkgs.sudo}/bin/sudo -u ${postgres.superUser} ${postgres.package}/bin/psql -c "ALTER ROLE meguca WITH PASSWORD '${cfg.password}';" || true
-        ${pkgs.sudo}/bin/sudo -u ${postgres.superUser} ${postgres.package}/bin/createdb -T template0 -E UTF8 -O meguca meguca || true
+        ${pkgs.sudo}/bin/sudo -u ${postgres.superUser} ${postgres.package}/bin/createuser \
+          -SDR meguca || true
+        ${pkgs.sudo}/bin/sudo -u ${postgres.superUser} ${postgres.package}/bin/psql \
+          -c "ALTER ROLE meguca WITH PASSWORD '$(cat ${cfg.passwordFile})';" || true
+        ${pkgs.sudo}/bin/sudo -u ${postgres.superUser} ${postgres.package}/bin/createdb \
+          -T template0 -E UTF8 -O meguca meguca || true
+
+        cat > /run/meguca/start << EOF
+#!/bin/sh
+cd /run/meguca
+${pkgs.meguca}/bin/meguca -d "$(cat ${cfg.postgresArgsFile})"\
+${optionalString (cfg.reverseProxy != null) " -R ${cfg.reverseProxy}"}\
+${optionalString (cfg.sslCertificate != null) " -S ${cfg.sslCertificate}"}\
+${optionalString (cfg.listenAddress != null) " -a ${cfg.listenAddress}"}\
+${optionalString (cfg.cacheSize != null) " -c ${toString cfg.cacheSize}"}\
+${optionalString (cfg.compressTraffic) " -g"}\
+${optionalString (cfg.assumeReverseProxy) " -r"}\
+${optionalString (cfg.httpsOnly) " -s"} start
+rm -f /run/meguca/start
+EOF
+
+        chmod 700 /run/meguca/start
       '';
 
       serviceConfig = {
-        PermissionsStartOnly = true;
         Type = "forking";
-        User = "meguca";
-        Group = "meguca";
```

Comment:
Why did you removed the `meguca` user here?

Reply:
The meguca user is no longer necessary, since it's running in /run/.
Actually, it was never really necessary, I don't know why I added it to begin with.
If it's part of some standard, I'll re-add.

Reply:
Actually never mind, I'll re-add it, since being able to configure the directory in which meguca starts in would be useful.

Reply:
But then the server would run as root, right? If it can be avoided network-facing services should not run as root.